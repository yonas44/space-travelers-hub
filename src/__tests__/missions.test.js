import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MissionsPage from '../components/Missions';
import store from '../redux/configureStore';

describe('Missions loads', () => {
  test('the component renders', () => {
    render(
      <Provider store={store}>
        <MissionsPage />
      </Provider>,
    );
    expect(screen.getByText(/Mission/)).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const renderComponents = renderer
      .create(
        <Provider store={store}>
          <MissionsPage />
        </Provider>,
      )
      .toJSON();
    expect(renderComponents).toMatchSnapshot();
  });
});
