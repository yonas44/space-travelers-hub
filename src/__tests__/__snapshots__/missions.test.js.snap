// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions loads matches snapshot 1`] = `
<div
  className="missions-holder"
>
  <table
    className="table-mission"
  >
    <thead
      className="missions-head"
    >
      <tr>
        <td
          className="text"
        >
          Mission
        </td>
        <td
          className="text"
        >
          Description
        </td>
        <td
          className="text"
        >
          Status
        </td>
        <td
          className="text"
        >
           
        </td>
      </tr>
    </thead>
    <tbody
      className="missions-body"
    />
  </table>
</div>
`;
